synapse/storage/databases/main/registration.py

```diff
@@ -1727,10 +1727,6 @@ def __init__(
             "refresh_tokens_device_index"
         )
 
-        self.db_pool.updates.register_background_update_handler(
-            "user_threepids_grandfather", self._bg_user_threepids_grandfather
-        )
```

Comment:
If we're happy to remove this update, we should probably make it a no-op so that when a homeserver upgrades from an ancient version to today's, it doesn't fail on this non-existent job. That's because https://github.com/matrix-org/synapse/blob/7ffddd819c2e0e7edff141e987372205894084b6/synapse/storage/schema/main/delta/53/user_threepid_id.sql#L28-L29 is a schema delta which inserts an entry into the background job queue, but you've just removed the job to which it refers.

Reply:
If it is pre 1.0.0 it might be OK to just remove?

Reply:
Purely out of curiosity: in general, why use no-op background updates vs adding another schema delta to drop an irrelevant background update job?

Reply:
hrm, that's a great question!

I think either solution would be fine. But since we've already got no-ops, there's something to be said for consistency?

Reply:
That's a valid point though; I didn't even think about using deletes in schema deltas for **removed** background jobs (rather than a few cases where we've deleted & re-inserted to prevent duplicates).
I personally don't like the no-ops as it feels like a monotonically increasing amount of cruft that we carry around in our handlers until the end of time, whereas at least the deltas feel more self-contained.
That said, hard to argue with consistency.

Reply:
Great thanks for answering, all! I've changed it to a no-op, I agree that it is hard to argue with consistency.